lib/Doctrine/ORM/Tools/EntityGenerator.php

```diff
@@ -1322,9 +1322,17 @@ protected function generateEntityFieldMappingProperties(ClassMetadataInfo $metad
                 continue;
             }
 
+            $defaultValue = '';
+            if (isset($fieldMapping['options']['default'])) {
+                if ($fieldMapping['type'] === 'boolean' && $fieldMapping['options']['default'] === '1') {
```

Comment:
I'd rather say that the default value should be passed through the DBAL type conversion system before being used in default value mapping, so this patch fixes the symptom, not the cause of the issue.

Reply:
Thanks for your reply!  
Okay. I'll try to fix the DBAL type conversion system.

Reply:
This means that this patch is unacceptable, right?

Reply:
DBAL contains the logic of the different types that are also used for the ORM entities. It should provide you with the functionality to get `true`.

Reply:
The mentioned API can be accessed this way:

```php
$defaultValue = Type::get($fieldMapping['type'])->convertToPhpValue($fieldMapping['options']['default']);
$defaultValue = ' = ' . var_export($defaultValue, true);
```

Reply:
Thanks for your comment.  

The `convertToPHPValue` method is had to pass `AbstractPlatform` value as a second argument parameter, isn't it?  
https://www.doctrine-project.org/api/dbal/2.8/Doctrine/DBAL/Types/Type.html#method_convertToPHPValue

I don't know how to get a `AbstractPlatform` value in `EntityGenerator` class.